strawberryfields/apps/data/feature.py

```diff
@@ -0,0 +1,148 @@
+        unitData (array): array of orbits/events used to construct the feature vectors
+        matData (array): array of adjacency matrices of graphs for which feature vectors
+            were calculated
+    """
+
+    @property
+    @abstractmethod
+    def _data_filename(self) -> str:
+        """Base name of files containing the data stored in the ``./FeatureData/`` directory.
+
+        For each dataset, feature vectors, information on what orbits/events are used to construct
+        these feature vectors and the corresponding adjacency matrices are provided in three
+        separate files.
+
+        Array of feature vectors should be provided in a text file. For ``_data_filename = "example"``,
+        feature vectors should be stored in ``./FeatureData/example_fv.txt``.
+
+        Array of corresponding adjacency matrices should be provided as a ``scipy.sparse.csr_matrix``
+        saved in ``.npz`` format. For ``_data_filename = "example"``, adjacency matrices
+        are stored in ``./FeatureData/example_mat.npz``."""
+        pass
+
+    @property
+    @abstractmethod
+    def _unit(self) -> str:
+        """Signifies the unit of construction of feature vectors; "orbits" or "events". This is
+        used with ``_data_filename`` to retrieve data stored in the ``./FeatureData/`` directory.
+
+        If ``_unit = "orbits"``, list of orbits used should be provided in a text file. Hence, for
+        ``_data_filename = "example"``, list of orbits should be stored in
+        ``./FeatureData/example_orbits.txt``.
+
+        If ``_unit = "events"``, list of events used should be provided in a text file. Hence, for
+        ``_data_filename = "example"``, list of events should be stored in
+        ``./FeatureData/example_events.txt``."""
+        pass
+
+    def __init__(self):
+        self.featuresData = np.loadtxt(DATA_PATH + self._data_filename + "_fv.txt")
+        self.unitData = np.loadtxt(DATA_PATH + self._data_filename + self._unit + ".txt")
+        self.matData = scipy.sparse.load_npz(DATA_PATH + self._data_filename + "_mat.npz").toarray()
+        self.num_fv, self.dim_fv = self.featuresData.shape
+
+    def __iter__(self):
+        return self
+
+    @property
+    @abstractmethod
+    def n_mean(self) -> float:
+        pass
+
+    @property
+    @abstractmethod
+    def threshold(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def method(self) -> str:
+        pass
+
+
+class QM9_Exact(FeatureDataset):
+    """ Exactly calculated feature vectors of 1000 randomly chosen molecules from the
+    ` QM9 dataset <http://quantum-machine.org/datasets/>`__ used in [ref] are provided.
+    Coulomb matrices are used as adjacency matrices in this case.
+    """
+
+    _data_filename = "QM9"
+    _unit = "orbits"
+    n_mean = 8
+    threshold = True
+    method = "exact"
+
+
+class QM9_MC(FeatureDataset):
+    """ Monte-Carlo estimated feature vectors of the same 1000 randomly chosen molecules from the

```

Comment:
When viewed in isolation, it won't be clear what we mean by "the same". How about adding a clarifying sentence at the end instead?